controllers/ocsinitialization/ocsinitialization_controller.go

```diff
@@ -403,11 +412,18 @@ func (r *OCSInitializationReconciler) getEnableTopologyKeyValue() string {
 
 func (r *OCSInitializationReconciler) getTopologyDomainLabelsKeyValue() string {
 
-	// return value from the internal storagecluster as failureDomain is set only in internal cluster
 	for _, sc := range r.clusters.GetStorageClusters() {
-		if !sc.Spec.ExternalStorage.Enable && sc.Status.FailureDomainKey != "" {
+		if !sc.Spec.ExternalStorage.Enable {
```

Comment:
lets add a check here if it is replica 1 then only pass it.

Reply:
exactly and I don't know how the ocsinitialization will know that. As we can get the info only from the json blob
So we are adding it with `(r *OCSInitializationReconciler) Reconcile(`  does it calls after `func (r *StorageClusterReconciler) Reconcile(ctx`, or they can be in parallel.

Reply:
if they are in parallel we need to separately do the unmarshal of the secret, or we should keep this update with the normal `externalreource` call  where we create the storageclass

Reply:
ocs initialization will know about replica-1 by the presence of the replica-1 storageclass, and accordingly, it will set the values.